sympy/printing/tests/test_lambdarepr.py

```diff
@@ -101,9 +98,8 @@ def test_piecewise():
     )
     l = lambdarepr(p)
     eval(h + l)
-    assert l == ("((1) if (x >= 1) else (((2) if (x >= 2) else (((3) if "
-        "(x >= 3) else (((4) if (x >= 4) else (((5) if (x >= 5) else (((6) if "
-        "(True) else None)))))))))))")
+    assert l == "((1) if (x >= 1) else (2) if (x >= 2) else (3) if (x >= 3)"\
+                        " else (4) if (x >= 4) else (5) if (x >= 5) else (6))"
```

Comment:
I like that this does not produce tons of parentheses at the end. But since you changed the handling of parentheses, I suggest adding a test where they matter: one Piecewise inside of another.  Like `Piecewise((Piecewise((1, x > 0), (2, True)), y > 0), (3, True))`